packages/playwright-utils/src/screenshots/index.tsx

```diff
@@ -0,0 +1,139 @@
+import type { Locator } from "@playwright/test";
+import type { JSX } from "vue/jsx-runtime";
+import ScreenshotMatrix from "./ScreenshotMatrix.vue";
+import type { MatrixScreenshotTestOptions, UseMatrixScreenshotTestOptions } from "./types";
+import { escapeGridAreaName } from "./utils";
+
+/**
+ * Creates a screenshot utility that can be used to capture matrix screenshots.
+ * Useful for capturing a single screenshot/image that contains multiple variants of a component.
+ */
+export const useMatrixScreenshotTest = ({
+  expect,
+  test,
+  defaults,
+}: UseMatrixScreenshotTestOptions) => {
+  const executeMatrixScreenshotTest = async <TColumn extends string, TRow extends string>(
+    options: MatrixScreenshotTestOptions<TColumn, TRow>,
+  ) => {
+    test(`${options.name}`, async ({ mount, page, browserName, makeAxeBuilder }) => {
+      // limit the max timeout per permutation
+      const timeoutPerScreenshot = 25 * 1000;
+      test.setTimeout(options.columns.length * options.rows.length * timeoutPerScreenshot);
+
+      /**
+       * Mounts the given element, captures a screenshot and returns and HTML `<img />` containing the captured screenshot.
+       */
+      const getScreenshot = async (element: JSX.Element, column: TColumn, row: TRow) => {
+        await page.getByRole("document").focus(); // reset focus
+        await page.getByRole("document").hover({ position: { x: 0, y: 0 } }); // reset mouse
+        await page.mouse.up(); // reset mouse
+
+        const component = await mount(element);
+        await options.beforeScreenshot?.(component, page, column, row);
+
+        const screenshot = await component.screenshot({ animations: "disabled" });
+
+        // some browser (e.g. safari) have different device resolutions which would cause the screenshot
+        // to be twice as large (or more) so we need to get the actual size here to set the correct image size below
+        // see (`scale` option of `component.screenshot()` above)
+        const box = await component.boundingBox();
+
+        // accessibility tests
+        const axeBuilder = makeAxeBuilder();
+        const disabledAccessibilityRules = [
+          ...(defaults?.disabledAccessibilityRules ?? []),
+          ...(options.disabledAccessibilityRules ?? []),
+        ];
+
+        if (disabledAccessibilityRules.length) {
+          axeBuilder.disableRules(disabledAccessibilityRules);
+        }
+
+        const accessibilityScanResults = await axeBuilder.analyze();
+        expect(
+          accessibilityScanResults.violations,
+          `should pass accessibility checks for ${column} ${row}`,
+        ).toEqual([]);
```

Comment:
Imho we should remove axe-core as a default dependency/include as it addresses a different concern and could be called as well in the `afterScreenshot` hook.

Reply:
I refactored the utility with a new `hooks` pattern, what do you think?

Reply:
Looks great! Amazing work :)